tables/audisp/src/audispservice.cpp

```diff
@@ -49,6 +54,9 @@ Status AudispService::exec(std::atomic_bool &terminate) {
   auto &socket_events_table_impl =
       *static_cast<SocketEventsTablePlugin *>(d->socket_events_table.get());
 
+  auto &file_events_table_impl =
+      *static_cast<FileEventsTablePlugin *>(d->file_events_table.get());
```

Comment:
You didn't start this pattern here but if we ever by accident leave these fields uninitialized we enter UB land. We should at least `assert(d->file_events_table)` before these derefercing operations, ideally for all fields we create handles of.

Reply:
The class follows RAII and the constructor will throw an exception if the tables can't be allocated when the AudispService object is created.

Reply:
I am marking this resolve. Holler if wrong.